modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/fork/ForkDetect.scala

```diff
@@ -0,0 +1,48 @@
+package org.tessellation.sdk.infrastructure.fork
+
+import cats.Monad
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+
+import org.tessellation.schema.SnapshotOrdinal
+import org.tessellation.schema.trust.TrustScores
+import org.tessellation.sdk.domain.fork.{ForkDetect, ForkInfo, ForkInfoMap}
+import org.tessellation.security.hash.Hash
+
+object ForkDetect {
+
+  def make[F[_]: Monad](
+    getTrustScores: F[TrustScores],
+    getForkInfo: F[ForkInfoMap]
+  ): ForkDetect[F] = new ForkDetect[F] {
+
+    def getMajorityFork: F[ForkInfo] =
+      getTrustScores.flatMap { trust =>
+        val scores = trust.scores.filter { case (_, s) => s > 0.0 }
+
+        getForkInfo.map { info =>
```

Comment:
```suggestion
      (getTrustScores, getForkInfo).mapN { case (trust, info) =>
        val scores = trust.scores.filter { case (_, s) => s > 0.0 }
```
If you take this suggestion than also use suggestion re. the import

Reply:
Changed to

```
      (getTrustScores, getForkInfo).mapN {
```